pos_frequency_pipeline.py

```diff
@@ -2,7 +2,62 @@
 Implementation of POSFrequencyPipeline for score ten only.
 """
 
+import json
+from pathlib import Path
+import re
+
+from pymystem3 import Mystem
+
+from constants import ASSETS_PATH
+from pipeline import CorpusManager
+from visualizer import visualize
+
 
 class POSFrequencyPipeline:
     def __init__(self, assets):
+        self.assets = assets
+        self.processed_text = ''
+        self.pos_freq = {}
+
+    def run(self):
+        """
+        Runs pipeline process scenario
+        """
+        articles = self.assets.get_articles()
+        for proc_article in articles.values():
+            self.processed_text = proc_article.get_raw_text()
+            self.pos_freq.update(self._process())
+            self._write_freq_meta(proc_article.article_id)
+            visualize(statistics=self.pos_freq,
+                      path_to_save=Path(ASSETS_PATH) / f"{proc_article.article_id}_image.png")
+
+    def _process(self):
+        """"
+        Counts frequencies for parts of speech in a text
+        """
+        result = Mystem().analyze(self.processed_text)
```

Comment:
the pipeline works on top of the processed dataset and does not perform analysis. Single Responsibility principle (!)